File: leetcode/861. Score After Flipping Matrix/main.go
package main

func maxInt(a, b int) int {
	if a < b {
		return b
	}

	return a
}

func matrixScore(grid [][]int) int {
	n, m := len(grid), len(grid[0])

	res := (1 << (m - 1)) * n

	for i := 1; i < m; i++ {
		val, set := 1<<(m-1-i), 0
		for j := 0; j < n; j++ {
			if grid[j][i] == grid[j][0] {
				set++
			}
		}

		res += max(set, n-set) * val
	}

	return res
}

File: leetcode/861. Score After Flipping Matrix/main.py
from typing import List


class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        rows, cols = len(grid), len(grid[0])

        res = (1 << cols - 1) * rows

        for c in range(1, cols):
            val, cnt = (1 << (cols - 1 - c)), 0
            for r in range(rows):
                if grid[r][c] != grid[r][0]:
                    cnt += 1

            res += max(cnt, rows - cnt) * val

        return res
